src/taxman.py

```diff
@@ -61,324 +78,606 @@ def __init__(self, book: Book, price_data: PriceData) -> None:
+
+                if isinstance(sc.op, tr.Deposit):
+                    # Raise a warning when a deposit link is missing.
+                    log.warning(
+                        f"You sold {sc.op.change} {sc.op.coin} which were deposited "
+                        f"from somewhere unknown onto {sc.op.platform} (see "
+                        f"{sc.op.file_path} {sc.op.line}). "
+                        "A correct tax evaluation is not possible! "
+                        "For now, we assume that the coins were bought at "
+                        "the timestamp of the deposit."
+                    )
+
+                self._evaluate_sell(op, sc)
+
+    def _evaluate_taxation_GERMANY(self, op: tr.Operation) -> None:
+        report_entry: tr.TaxReportEntry
+
+        if isinstance(op, (tr.CoinLend, tr.Staking)):
+            # TODO determine which coins get lended/etc., use fifo if it's
+            # unclear. it might be worth to optimize the order
+            # of coins given away (is this legal?)
+            # TODO mark them as currently lended/etc., so they don't get sold
+            pass
+
+        elif isinstance(op, (tr.CoinLendEnd, tr.StakingEnd)):
+            # TODO determine which coins come back from lending/etc. use fifo
+            # if it's unclear; it might be nice to match start and
+            # end of these operations like deposit and withdrawal operations.
+            # e.g.
+            # - lending 1 coin for 2 months
+            # - lending 2 coins for 1 month
+            # - getting back 2 coins from lending
+            # --> this should be the second and third coin,
+            #     not the first and second
+            # TODO mark them as not lended/etc. anymore, so they could be sold
+            # again
+            # TODO lending/etc might increase the tax-free speculation period!
+            # TODO Add Lending/Staking TaxReportEntry (duration of lend)
+            # TODO maybe add total accumulated fees?
+            #      might be impossible to match CoinInterest with CoinLend periods
+            pass
+
+        elif isinstance(op, tr.Buy):
+            # Buys and sells always come in a pair. The buying/receiving
+            # part is not tax relevant per se.
+            # The fees of this buy/sell-transaction are saved internally in
+            # both operations. The "buying fees" are only relevant when
+            # detemining the acquisition cost of the bought coins.
+            # For now we'll just add our bought coins to the balance.
+            self.add_to_balance(op)
+
+        elif isinstance(op, tr.Sell):
+            # Buys and sells always come in a pair. The selling/redeeming
+            # time is tax relevant.
+            # Remove the sold coins and paid fees from the balance.
+            # Evaluate the sell to determine the taxed gain and other relevant
+            # informations for the tax declaration.
+            sold_coins = self.remove_from_balance(op)
```

Comment:
Should this ignore `config.FIAT`? I'm getting the error `Not enough EUR in queue to sell`